### DISCUSSION TITLE
Welcome to PiFire Discussions!

### DISCUSSION THREAD

**Discussion by nebhead** [nebhead/PiFire#15] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by faisalhusain007** [comment]
Hey nebhead, well done!  Wow I don't often comment, but sometimes very rarely I do. Your PiFire is so polished and exceptional. I am testing the hardware and software right now and I think you have done an awesome job. Right now I have everything on my workbench, (Ok my dining table come workbench). The info and documentation on your git is very thorough, thank you for that, I really appreciate it. . Some things I have changed in my hardware are the use of SSR relay boards as they save space and use less current on the switching side, downside is they only switch AC and only have 1 switch position. This ok as I intend to replace the original controller. I will work on a 3d printed enclosure to place PiZero and SSR board and PSU, Resistor Divider and  Screen in the original space. I have used a small 2.5A usb charger 40mmX40mm. I would need to add some push buttons to the Pi i/o pins and some menu items for basic push button operations will keep you posted.

I was wondering what your thoughts were about PID vs standard cycle, some people say that standard non PID gives a better smoke flavour, PID by its very nature gives very little to no smokey flavour, I am not sure as I have yet to try PID on my grill. Do you think it would be useful to use cycles like the normal controller, I know it can cause massive swings in Temp, I haven't  tried it myself (PID). Perhaps have a module that can be selected from the web interface to choose between PID and non PID ?

Anyhow Thanks again for sharing everything with the community you have done an awesome job so far.

**Comment by nebhead** [comment]
First and foremost, thank you so much @faisalhusain007!  It really makes it all worth it when I hear people are able to use my projects.  Because I also consider myself a novice at much of this stuff, I often hesitate sharing these things with the world - but it's comments like these that give me the most confidence and joy.  

I'm actually really curious about the SSR relay boards, what you might have used, and how it goes.  I have been using the cheap relay boards from Amazon because they are, well, cheap.  But really hadn't considered any thing else up until now.  Might be nice to have a smaller footprint and lower current draw.  Also, if you 3D print a case, that would be really neat to see. I also have a 3D printer and had considered printing something up, but never got around to it.  Instead just decided to use the pre-made project box which was definitely weather-proof.  It was pretty expensive though - so a 3D printed alternative would be nice.  

As for PID versus standard cycle, this a very good question that I hadn't been asked yet.  To be honest, I don't have any good scientific way to measure the level of smokiness in any of my cooks, but what I do feel is that if there is a difference (at lower temperatures below 200F), that it's really hard to tell.  

With that said, in this project, I've made the "Smoke" mode be a standard cycle (non-PID), just like the Traeger controller, with a P-Setting to offset for ambient temperatures.  I find this works quite well for smoking most everything - and I use this for smoking ribs (at least for the first stage.  Then, when I want to get into the cooking temperature ranges, I generally want to maintain tighter temperature tolerances - and that's when the PID controller comes into play.  The PID is borrowed from the SmokerPi project, and seems to do a very nice job of maintaining temperature - however I have seen some other PID projects on GitHub that add a few more bells and whistles, i.e. preventing excessive windup.  

I've seen that with Traeger's newer controllers, they have something called "Super Smoke", which I understand from some comments on the internet, is modulating the auger fan somehow.  So there may be some combination of the standard cycle and fan modulation that could be achieved to generate more smoke.  However, this would require much more experimentation, and some way to measure the smoke output.  It would be cool to do, but thus far I've been happy with my results.  

Anyway, thank you again for your kind words, good luck with your build and please share your updates here!

-Ben.

**Reply by faisalhusain007** [reply]
The SSR board was fairly cheap, around $5 here are the specs for it https://my.cytron.io/p-4-channels-solid-state-relay-module-low-trigger?search=ssr&description=1&src=search.list.

I think it's difficult to vary the speed of the AC fan, however I was thinking that we could pulse the SSR therefore changing the speed. I am waiting for a spare fan to arrive and test it out. (The SSR on the AC load side uses a TRIAC)

**Reply by nebhead** [reply]
Oh very nice!  So, am I correct in seeing that it only has two terminals for each relay?  That's one difference for me that might not work out so well, given that I'm using the normally open and normally closed connections to allow for my Traeger to still function when I'm not using the PiFire.  (however, I haven't used the Traeger controller in months)  So, if you are completely replacing the existing controller, these will work just fine.  

For the fan, I was actually thinking instead of doing a PWM type thing to vary the fan speed, more of a cycle of a few seconds on and a few seconds off to see if that modulates the smoke at all... and/or the temperature.  I suspect that would lower the temperature a bit, so a longer pellet auger cycle might be needed.

**Reply by faisalhusain007** [reply]
Yes they only have 2 terminals, and since they are using TRIAC's only good for switching AC loads. You can get SSR's with same outputs as analogue with 3 outputs no-com-nc, so you could use with your existing setup.

My idea is to use PWM on the fan SSR, since the SSR is effectively an isolated TRIAC we can switch it using PWM to hopefully adjust the speed of the fan. Waiting on the spare fan to test it out.

**Reply by nebhead** [reply]
Excellent!  I am eager to hear how it goes.  Might be very intriguing to see if we can actually PWM an AC fan and see how that might effect the smoke output.  Let me know once you get your spare fan in hand.

**Comment by faisalhusain007** [comment]
Preliminary design for the front panel 

![Traeger_Pellet_Grill_controller_2020-Dec-30_02-58-01PM-000_CustomizedView4774087445](https://user-images.githubusercontent.com/36872476/103394446-947d0980-4b63-11eb-991f-53f3b7f302b0.png)

![4Traeger_Pellet_Grill_controller_2020-Dec-31_04-16-43AM-000_CustomizedView10384547251](https://user-images.githubusercontent.com/36872476/103394461-b1b1d800-4b63-11eb-9be9-6552489774cf.png)
![3Traeger_Pellet_Grill_controller_2020-Dec-31_04-16-06AM-000_CustomizedView20195807495](https://user-images.githubusercontent.com/36872476/103394462-b4acc880-4b63-11eb-870e-e59d93ea608a.png)
![2Traeger_Pellet_Grill_controller_2020-Dec-31_04-15-24AM-000_CustomizedView42085507854](https://user-images.githubusercontent.com/36872476/103394463-b5455f00-4b63-11eb-9ffa-1d8248314f8f.png)
![1Traeger_Pellet_Grill_controller_2020-Dec-30_03-10-09PM-000_CustomizedView10845961309](https://user-images.githubusercontent.com/36872476/103394464-b5ddf580-4b63-11eb-89c0-e47677ef94e4.png)

**Reply by faisalhusain007** [reply]
Still waiting on some components to finalize the dimensions and build but will upload everything when finished here.

**Reply by nebhead** [reply]
This is amazing!  I might have to print this up myself and start just using this instead of the Traeger controller.  Really nice work!

**Reply by nebhead** [reply]
What software are you using to model this?  Just curious as I'm just starting to make the leap from Tinkercad to bigger better software.

**Reply by faisalhusain007** [reply]
> This is amazing! I might have to print this up myself and start just using this instead of the Traeger controller. Really nice work!

Still waiting for mono jacks and tactile button switches to finalize dims

**Reply by faisalhusain007** [reply]
> What software are you using to model this? Just curious as I'm just starting to make the leap from Tinkercad to bigger better software.

Using fusion360, by no means am I an expert but there are plenty of tutorials and resources available.  Tinkercad is pretty good for quick design but I tend to use fusion now for most things I 3d print. Let me get everything finalized, I will share the files here once finished.

**Reply by faisalhusain007** [reply]
> This is amazing! I might have to print this up myself and start just using this instead of the Traeger controller. Really nice work!

Once I get everything finalized will share the files and design here

**Comment by faisalhusain007** [comment]
Quick test to see if the ssr are up to the job
looks like they are, didn't hook up my scope yet with the fan load


https://user-images.githubusercontent.com/36872476/103455798-271bd500-4d2b-11eb-8397-26a474ea1a24.mp4

**Reply by nebhead** [reply]
OK, I'm convinced.  This is amazing.  Just ordered my SSR relays on AliExpress now.  :)  (now I just have to wait a month for them to get here)

**Reply by faisalhusain007** [reply]
Did you order the board or ssr individually? The ssr on the board use the omron g3mb-202p here is a link to the data sheet. I have tested various freq and duty cycles , not much heat appears to be building up on ther ssr I will try and hook my scope up in the next few days and try and see what  and freq + duty cycles  gives smooth speed. https://www.alldatasheet.com/view.jsp?Searchword=G3mb-202p&gclid=Cj0KCQiA0MD_BRCTARIsADXoopa0UbvhW04M0EncbaQdMOcluaIdd5nhCXGwz9HSvCNrX_MXc-O_OtsaAiO6EALw_wcB

**Reply by nebhead** [reply]
Ordered the board from here: https://www.aliexpress.com/item/1005001621690700.html?spm=a2g0s.12269583.0.0.69fe5326cVXex3  

Not sure what actual relays are on this board, but I suppose we will find out. :)

**Reply by faisalhusain007** [reply]
On a side note the original controller on the traeger uses a st bta12 600b triac switched with an optical isolator.

**Reply by faisalhusain007** [reply]
Its using the same omron SSR, if you send me an address where you can receive mail, I will courrier from here in Malaysia should take less than a week, do let me know.

**Reply by nebhead** [reply]
Oh that's very generous of you, but I'll probably just wait - sometimes they ship a little faster.  Frankly speaking, as my day job starts up again next week I won't have as much time to work on this right now anyway.  

What part of Malaysia?  I've been to Penang once and it was lovely (despite the fact that it was for work).

**Reply by faisalhusain007** [reply]
I used to live in Penang in Batu Ferengi near miami beach. Now I live in Kuala Lumpur

**Reply by nebhead** [reply]
Very nice!  I've been to Batu Ferengi, to the night market and to the Hard Rock there.  Good memories and a good trip.

**Reply by faisalhusain007** [reply]
> Very nice! I've been to Batu Ferengi, to the night market and to the Hard Rock there. Good memories and a good trip.

Yep famous night market by the sea front. Havent been back in almost 10 years, I  think its probably changed a lot. Where did you work? Was it around byanlepas?

**Comment by faisalhusain007** [comment]
I have been playing about a bit with the code to include a PWM FAN button in manual.html and adding a few lines of code in grillplat_traeger.py, I have only succeeded in breaking the entire control when adding pwm code for the fan in grillplat_traeger.py . I was wondering if it's possible to add a slide scale similar to the 'Hold temperature' for Duty cycle 0 to 100% and an input for frequency. I have noted that the higher the frequency the hotter the motor gets, so need to do some work on good combinations of duty cycle, frequency and speed achieved. On my tachometer i measured just over 2500rpm on standard non pwm.

**Reply by nebhead** [reply]
Yep, we can do that.  Will require adding something to app.py as well, since it will be on the backend and we will likely also need to add something to the settings.json & control.json to indicate default and communicate the PWM to the control program.  Also, are you using HW PWM or SW PWM?  If we use the HW PWM, we might need to switch some pins around to accommodate.   I haven't looked at what pin(s) is being used for HW PWM recently.

**Reply by nebhead** [reply]
Tell you what, I'll take a crack at this and put the code on up the dev branch so you can test it out.

**Reply by faisalhusain007** [reply]
> Yep, we can do that. Will require adding something to app.py as well, since it will be on the backend and we will likely also need to add something to the settings.json & control.json to indicate default and communicate the PWM to the control program. Also, are you using HW PWM or SW PWM? If we use the HW PWM, we might need to switch some pins around to accommodate. I haven't looked at what pin(s) is being used for HW PWM recently.

Using software pwm so can use any pin, I think they're only 4 h/w pins so prefer to stick to software pwm

**Reply by faisalhusain007** [reply]
> Tell you what, I'll take a crack at this and put the code on up the dev branch so you can test it out.

Awesome, that would be great,  thank you.

**Reply by nebhead** [reply]
> Using software pwm so can use any pin, I think they're only 4 h/w pins so prefer to stick to software pwm

I'll take a look - if it's going to be too difficult to get HW PWM to work within the current setup, then I'll just stick with SW PWM.

**Reply by nebhead** [reply]
Long story short, I haven't had a chance to get started on making changes yet due to some other things I had to take care of today.  

However, I did just poke around on Reddit, YouTube, etc. and have learned more about the Super Smoke mode.  It seems like that in this mode, they are simply turning the fan on/off every 3 seconds.  So they aren't ramping it to 100 and down to 50% or anything fancy - so it doesn't seem to be a PWM thing.   [Video example of cycling here.](https://www.reddit.com/r/Traeger/comments/irbqbt/new_ironwood_650_is_this_fan_noise_normal/)  Also, lots of argument both ways regarding whether it makes a difference in smoke levels (also that it might require the vented back, instead of the smoke stack design).  

Regardless, I think it would be fun to play with PWM to see if we can make it work out, but it frankly might just be as simple as cycling the fan every three seconds (which is great because that's the cycle that I use for checking temps).  

Probably going to be a little more challenging to get it integrated than I thought initially, but I will try to get to it this week sometime.

**Reply by faisalhusain007** [reply]
> Long story short, I haven't had a chance to get started on making changes yet due to some other things I had to take care of today.
> 
> However, I did just poke around on Reddit, YouTube, etc. and have learned more about the Super Smoke mode. It seems like that in this mode, they are simply turning the fan on/off every 3 seconds. So they aren't ramping it to 100 and down to 50% or anything fancy - so it doesn't seem to be a PWM thing. [Video example of cycling here.](https://www.reddit.com/r/Traeger/comments/irbqbt/new_ironwood_650_is_this_fan_noise_normal/) Also, lots of argument both ways regarding whether it makes a difference in smoke levels (also that it might require the vented back, instead of the smoke stack design).
> 
> Regardless, I think it would be fun to play with PWM to see if we can make it work out, but it frankly might just be as simple as cycling the fan every three seconds (which is great because that's the cycle that I use for checking temps).
> 
> Probably going to be a little more challenging to get it integrated than I thought initially, but I will try to get to it this week sometime.

Really appreciate your help, no rush just like you said nice to have. I think the pulsing of the fan may work too, might be easier to implement. I think the idea is at lower temps we want less airflow to the firepot that should create more smoke let the wood pellets smolder ....

**Reply by nebhead** [reply]
Yep, I think pulsing the fan at certain cycles is the way to go.  Was thinking that there is probably a reason why they restrict to 165-225F ranges for this mode.  Thus, I think my plan will be to implement an on-off cycle (adjustable for now) between these temp ranges for smoke mode.  

Do you happen to know if you can do PWM frequencies less than 1 Hertz?  Like 0.15Hertz, such that if we have a 50% duty cycle it is 3 seconds on and 3 seconds off?  I suppose I could give it a try on the Pi that I have on hand, but if you have one setup already that would be cool to know.

**Reply by faisalhusain007** [reply]
Let me check it out and get back to you. In terms of SW PWM can set as duty freq on pi to 0.1 in theory. The other issue I have is I need to use a step down transformer to 120VAC but the electricity here is rated at 50Hz not 60Hz so I instantly lose about ~20% speed on the fan.  Checking the speck of the fan on the grill should be about 3000 RPM, but I only measured ~2500 RPM. Anyways let me test the freq and duty cycle and get back to you.

**Reply by faisalhusain007** [reply]
seems to work ok but havent hooked my scope up

https://user-images.githubusercontent.com/36872476/103601298-4ccff680-4f44-11eb-9bbe-02fe00c17636.mp4

**Reply by nebhead** [reply]
FYI... just implemented a very straight-forward on-off cycle for the fan which is adjustable from the settings.  Called it Smoke-Plus (a little marketing flair).  The commit is now in the development branch, but I haven't tested this on my real hardware - only in prototype mode.  

When in smoke mode, you can just click the smoke button (in the WebUI) again to enabled Smoke-Plus (and a little plus icon shows up in button the WebUI).  This should pulse the fan at a 50% duty cycle over 10 second cycles when the temperature is between 160F-220F.  

I'll try it when I get a chance.  But I wanted to get something basic implemented and up to test.

**Reply by faisalhusain007** [reply]
I will try and test it out soon. I have updated the case design from the previous but I have not printed the new design, will do it soon.

**Comment by faisalhusain007** [comment]
I did a print of the design above and installed it. Definitely a beta, some changes for the final to make life easier for maintenance. 
And other minor changes for better fit
![IMG-20210106-WA0005](https://user-images.githubusercontent.com/36872476/103687786-cc54d880-4fcb-11eb-94ed-4fbef2c6fa46.jpg)
![IMG-20210106-WA0007](https://user-images.githubusercontent.com/36872476/103687835-e1316c00-4fcb-11eb-82c3-514f117898dc.jpg)
![IMG-20210106-WA0008](https://user-images.githubusercontent.com/36872476/103687887-f60dff80-4fcb-11eb-804a-ae54e522e3a9.jpg)
![IMG-20210106-WA0006](https://user-images.githubusercontent.com/36872476/103687926-0920cf80-4fcc-11eb-8719-03886ebdb3df.jpg)

**Comment by nebhead** [comment]
Wow, that looks great!  Very clean indeed.  Nice job!

So are the two holes in the front for the probe inputs?  What are the other holes for?

**Reply by faisalhusain007** [reply]
Yes the 2 large holes are for the probes, the smaller for tactile buttons, but I will change this as it is difficult to mount the buttons there. Last night I mounted the 2.5mm jacks . I will revise the design and refine it after more testing

**Comment by faisalhusain007** [comment]
I have redesigned the case for better and easier fit for buttons and temp probes, bellow is a render


![Traeger_Pellet_Grill_controller_2021-Jan-07_07-18-32AM-000_CustomizedView49787425662](https://user-images.githubusercontent.com/36872476/103866838-5e123200-5101-11eb-8cfe-1e430df08cdd.png)

I have not printed out and tested it yet.

This is a render for front plate that sits flush, I would not use this design as it delt flimsy to me when I tried it before.
![Traeger_Pellet_Grill_controller_2021-Jan-07_07-19-56AM-000_CustomizedView49787425662](https://user-images.githubusercontent.com/36872476/103867029-a598be00-5101-11eb-8589-0e43ec439c05.png)

Here are renders of the various parts

![Traeger Pellet Grill controller v1](https://user-images.githubusercontent.com/36872476/103868095-64a1a900-5103-11eb-8407-776c8cc03a8f.png)

![Traeger Pellet Grill controller v1 logo](https://user-images.githubusercontent.com/36872476/103868170-7c792d00-5103-11eb-8aaa-a2f385ec6b5e.png)

Finally a short animation

https://user-images.githubusercontent.com/36872476/103868256-9c105580-5103-11eb-98c7-57d87186c3e3.mp4


Bellow is a link to the various parts as STL`s still not final and I have not printed and tested the final designs so consider it WIP. I have linked to thingerverse https://www.thingiverse.com/thing:4713318

**Reply by nebhead** [reply]
Wow, this looks amazing!!  I can't wait to print this up myself and get it installed.  I love the buttons and really like the idea of adding the logo.  So cool!  Really good job!

**Reply by faisalhusain007** [reply]
Actually I need to update the buttons and the logo will do it tomorrow and upload just to make it easier to install.

**Reply by faisalhusain007** [reply]
Buttons and logo have been updated uploaded to thingiverse

**Reply by faisalhusain007** [reply]
I have printed out everything and made some minor mods after fitting etc. All new files have been updated on thingiverse and are ready to go.

![WhatsApp Image 2021-01-09 at 23 43 34 (1)](https://user-images.githubusercontent.com/36872476/104095942-99b12580-52d4-11eb-99d8-637c89f8a15d.jpeg)
![WhatsApp Image 2021-01-09 at 23 43 34](https://user-images.githubusercontent.com/36872476/104095948-9cac1600-52d4-11eb-8f9f-03bbb83359e4.jpeg)

**Reply by nebhead** [reply]
That's starting to look really nice!  I suppose I should work more on the button interface so we can get those buttons working. :)  I really like what you've done with those.  So, how did you get the surface to be textured like that?

**Reply by faisalhusain007** [reply]
I used a Magnetic Double Sided Textured + Smooth PEI Spring Steel Sheet
![Screenshot_20210110-113603_Chrome](https://user-images.githubusercontent.com/36872476/104116967-59938680-5358-11eb-9c1d-381608b57618.jpg)

http://3dgadgets.my/print-bed/1098-magnetic-double-sided-textured-smooth-pei-spring-steel-sheet.html

**Comment by faisalhusain007** [comment]
I am having some  issues the grill never gets above 130f, it smolders and smokes like anything. I checked the P setting tried factory 2 then 3 made it worse. right now I have it on 0, when looking at how many pellets drop in the firepot doesnt look like many. not sure if I should tweak the 15s on auger setting. I should hook the original controller back on it and check. But last I used it was ok.

**Reply by faisalhusain007** [reply]
I havent been able to check the smoker + feature because I cant seem to get to 160f I have attached csv of the temp over an hour

[export.csv.txt](https://github.com/nebhead/PiFire/files/5782335/export.csv.txt)

**Reply by nebhead** [reply]
Just curious, if it worked prior to the modifications that I did for Smoke-Plus?  I'd like to rule that out, if it worked previously. 

Two other things you might want to check:

1. Double check they cycle timings.  You can turn on debug mode in the admin screen and then run a cycle, and check the debug.log to see what the cycles timings look like.  If they are matching up roughly with the 15 second auger on time, and variable off time (based on P-Mode).  I kind of doubt there are timing issues, but it's worth a check.  Also, when you are done with debug mode, it's probably best to turn it off since it generates a lot of output.    
2. Make sure your temperature probe is working properly.  This is the main variable that might be off with your temperature probe versus mine.  It might require some tuning to get right.   If you have another thermometer or probe that you can drop into the barrel, to confirm the temperatures, that might be helpful.  If you see a large disparity between the temperatures that you get from another probe to the grill probe then we might need to tune your profile for that.  I suspect that _this_ is the issue here.  
3.  Maybe your auger is slower?  That certainly is a possibility, but these things are pretty standard so I would hope that's not the case.  

It takes my grill about 20-30 minutes to get up to 160 (at least that's what I'm seeing from my last cook).  That's based on an ambient temperature around 55F (which is a bit cold for my area) and a P-Mode setting of 2.  So it sounds like you are waiting long enough... 

I really think it's got to be the grill probe.  They generally are pretty low quality probes and may vary in resistance from probe to probe, so that could be affecting things.  Luckily, I have a tuning tool in the settings page (in the profile editor section) that will help us get you tuned up.

**Reply by faisalhusain007** [reply]
There is a big difference between the grill probe and the one on the traeger about 12f out I did use the tuning tool on the other probes from my inkbird and they are calibrated really well (one of which is in the grill as a check) I found the long RTD probe to be slower in updating temperatures compared to my inkbird ones. I have debug turned on and looks ok. Let me check the tuning on the grill probe.

**Reply by faisalhusain007** [reply]
![image](https://user-images.githubusercontent.com/36872476/103927243-043b5780-5155-11eb-8384-76a3688e1d01.png)
Probe 1 is ambient probe 2 is in the grill

**Reply by nebhead** [reply]
Wow, you are very fast. :)  To be honest, 12F isn't too far off from what I've seen on these RTD probes.  But definitely it can be dialed in closer to a couple degrees difference.  They do tend to be slower in general, but I think that's just a property of the thermistor itself.  

All of this still wouldn't explain why you aren't seeing a higher smoke temperature.  Like you were thinking, maybe we either reduce the off cycle lower... or we increase the 15 second on cycle to 20 seconds or something.

**Reply by faisalhusain007** [reply]
I think I will increase the on time to 20 sec

**Reply by nebhead** [reply]
Let me know if you need any help making those changes.  Let me know how it goes.

**Reply by faisalhusain007** [reply]
Just in one place I think it's in control.py right change the on time from 15 to 20

**Reply by nebhead** [reply]
Yep, should just be in `control.py`

Edit the OnTime variable: 
```
	if (mode == 'Startup' or 'Smoke' or 'Reignite'):
		OnTime = 15		#  Auger On Time
		OffTime = 45 + (settings['cycle_data']['PMode'] * 10) 	#  Auger Off Time
		CycleTime = OnTime + OffTime 	#  Total Cycle Time
		CycleRatio = OnTime / CycleTime #  Ratio of OnTime to CycleTime

```

**Reply by faisalhusain007** [reply]
Tuned the probe and upped the ontime to 20 seconds much better 

![image](https://user-images.githubusercontent.com/36872476/104026495-f6431080-5200-11eb-8f68-5f447f012472.png)

Thank you for the help

**Reply by nebhead** [reply]
Outstanding!  

I am planning to test out the Smoke Plus mode this weekend.  Unfortunately, I'm dangerously low on pellets, so I'm going to pick some up today, so I can do some cooking tomorrow.  I'm hopefully that this new mode will work out.   BTW... in settings you can lower the minimum temperature that Smoke Plus will start cycling the fan, to something around 140F just to see if it does anything for smoke output.  

Also working on another new feature that I'm excited to share - but it's far from ready yet.  :)

**Reply by faisalhusain007** [reply]
Sorry forgot to tell you I tested the smoke + feature yesterday and it works really well, I set it to kick on 140 so I could test it. Creates large clouds of smoke.  I like the that you can configure the fan cycle time.

**Reply by faisalhusain007** [reply]
What's the new feature?

**Reply by nebhead** [reply]
>Sorry forgot to tell you I tested the smoke + feature yesterday and it works really well, I set it to kick on 140 so I could test it. Creates large clouds of smoke. I like the that you can configure the fan cycle time.

That's great news!!  I'm excited to try it out myself this weekend.  Was thinking I might add an option to change the duty cycle as well, so it's not just set at 50% all the time.  

>What's the new feature?

I'm probably over-selling it, but I will keep it under wraps for now until I can unveil it. :)

**Comment by Mendingthings** [comment]
Moved this to the correct discussion thread I believe. 
Hi, I'm new to this project. I had the idea to do almost this exact same thing and found your project. I have been reading and following for a few days. I like all the ideas and what you are doing with this. I am an electronics buff, with some coding experience as well as 3d printing, CNC machining, and am a fusion 360 user for all my stuff. I don't claim to be an expert by any means but I am willing to help if needed with anything but you guys seem to have a good handle on what you are doing. I am considering continuing with my initial plan to build a similar project based around the ESP32. I guess we could use one of the varieties that run micro python and port this project into it. Either way, I hope you don't mind me tagging along to learn from you guys as well.

**Reply by nebhead** [reply]
Welcome to the project @Mendingthings!  

You're definitely welcome to help by providing your own designs, changes, ideas to this project.  And if you decide to build your own using the ESP32, then I'm happy to help with any questions or advice you may have along the way.  I'm also a bit of a novice, many of these topics, and have been using projects like this to help me learn along the way and am always eager to help those of us in the community with similar interests.  

Looking forward to good discussion!

**Reply by faisalhusain007** [reply]
Hey @Mendingthings I was also looking to do the same thing on an ESP32 and found this project @nebhead. Since he had such a polished project I opted for using his project. I have done many projects in the past on ESP's but this is my first using python, I haven't played much with micro python or python, but if you need my help I will try.

**Reply by Mendingthings** [reply]
Thanks, I plan to soak in a little more before I offer much. I need to get up to speed on all the great things this project has already done. The more I look the more I think "why re-invent the wheel" but I do like a challenge.
